lib/src/interactable_element.dart

```diff
@@ -32,10 +35,38 @@ InteractableElement parseInteractableElement(
   switch (element.localName) {
     case "a":
       interactableElement.href = element.attributes['href'];
```

Comment:
Too many abstraction levels in the body of 1 method.
You should break it down to smaller method/func

Infact, the code should do like this ( i will assume that your code logic is correct)

if(hasNoChild(interactableElement)) {
  underline(interactableElement);
} else {
   underlineChildTextElements(interactableElement);
}

Then, the method underlineChildTextElements will take care the recurrsive search & apply  the corresponding style to its children